benchexec/tools/ultimate.py

```diff
@@ -204,113 +188,125 @@ def _requires_ultimate_data(self, executable):
         # all versions before 0.1.24 do not require ultimatedata
         return not (int(major) == 0 and int(minor) < 2 and int(patch) < 24)
 
-    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits=None):
-        if rlimits is None:
-            rlimits = {}
-
-        self._uses_propertyfile = propertyfile is not None
-        if _OPTION_NO_WRAPPER in options:
-            # do not use old wrapper script even if property file is given
-            self._uses_propertyfile = False
-            propertyfile = None
-            options.remove(_OPTION_NO_WRAPPER)
+    def cmdline(self, executable, options, task, resource_limits):
+        data_model_param = get_data_model_from_task(
+            task,
+            {ILP32: ["--architecture", "32bit"], LP64: ["--architecture", "64bit"]},
+        )
+        combined_options = options
+        if data_model_param and not self._is_sublist(data_model_param, options):
+            combined_options += [data_model_param]
```

Comment:
Either change this line to `combined_options += ["--architecture", data_model_param]"` and make the values of the above dict to simple strings, or change this line to `combined_options += data_model_param` (though in the later case, `data_model_param` should really be named `data_model_params`.

This is the cause of the crash due to a nested list being contained in the result value.